### ISSUE TITLE
Migrate to Gatsby

### ISSUE THREAD

**Issue by kbariotis** [skgtech/devit#410]
Jekyll served us well but unfortunately, we've hit a point where development is not fun anymore and the tools right now seriously slow development. Moreover, Jekyll doesn't support modern mechanisms like code splitting and PWA features.

[Gatsby](http://gatsbyjs.org/) is a modern static site generator based on React and Webpack while it implements a GraphQL server to feed data. It provides most of the functionalities we are looking for out of the box. 

In order for us to do the migration though, there are is a thing we need to take care first and that is to remove the Bootstrap dependency which depends on a global layout stylesheet and on jQuery. Things we need to tackle are:

- [ ] Implement a Grid alternative, probably [react-flexbox-grid](https://github.com/roylee0704/react-flexbox-grid)
- [ ] Implement a responsive navigation top menu
- [ ] Implement an alternative to Bootstrap's tabs module

Other stuff might come out along the way and they will be stated here.

That would also be a great opprortuninty to get to know more Gatsby as it's a great and battle tested too.

@varemenos has already started working on the basis for the new [project here](https://github.com/varemenos/devit-gatsby)

/cc @modkaffes @joskar @alexspyr Thoughts?

**Comment by joskar** [comment]
Also I think this is a good time to rethink the modeling of the data.

**Comment by alexspyr** [comment]
I agree that Jekyll is giving us a hard time ( although I have limited experience with it). Also I like the idea of using React components 👍 What's the timeline for the migration?

**Comment by kbariotis** [comment]
No timeline, we have a website in place right now so nothing is rushing us.
It would be better to deprecate the old website at some point this year so
we can do any improvements and new features on the the new one.
